framework/src/parser/Parser.C

```diff
@@ -824,8 +828,12 @@ Parser::buildFullTree(const std::string & search_string)
           {
             pos = act_name.size();
 
+            // Remove <RESIDUAL> append for AD objects
+            std::string obj_name = moose_obj->first;
+            removeSubstring(obj_name, "<RESIDUAL>");
+
             if (!action_obj_params.collapseSyntaxNesting())
-              name = act_name.substr(0, pos - 1) + moose_obj->first;
+              name = act_name.substr(0, pos - 1) + obj_name;
```

Comment:
Good old trial and error got me to this only substitution. The dump file still has <RESIDUAL>, but autocomplete still works. Hopefully nothing else gets screwed up